doc/user/integrations/docker.md

```diff
@@ -0,0 +1,23 @@
+---
+title: "Using Docker"
+description: "Get details about using Materialize with Docker"
+menu:
+  main:
+    parent: 'integrations'
+---
+
+Many of our demos rely on Docker and Docker Compose to make it easy to deploy Materialize, along with any other infrastructure the demo needs.
+
+For the best experience using Docker, we recommend following the guidelines outlined here.
+
+### Increase Docker Memory
+
+Because many of our Docker-based demos leverage a large number of pieces of infrastructure, we recommend running Docker at least **8 GB** of memory.
+
+On macOS:
+
+1. Open **Docker for Mac**'s **Preferences** window.
+1. Click **Resources**.
+1. Click **Advanced**.
+1. Move the **Memory** slider to at least **8.00 GB**.
+1. Click **Apply & Restart**.
```

Comment:
I think this should mention that on Linux Docker will automatically share memory with the host machine, so as long as the host machine has more than 8GB of memory you'll be good to go.